backend/src/tasks.ts

```diff
@@ -0,0 +1,349 @@
+        const quiz = await hre.ethers.getContractAt('Quiz', args.address);
+        await setReward(hre, quiz, args.reward);
+    });
+
+// Send funds from signers account to quiz contract.
+task('fund')
+    .addPositionalParam('address', 'contract address')
+    .addPositionalParam('amount', 'amount in ROSE')
+    .setAction(async (args, hre) => {
+        const quiz = await hre.ethers.getContractAt('Quiz', args.address);
+        await fundContract(hre,
+            quiz,
+            args.amount
+        );
+    });
+
+// Send funds from quiz contract to specified address.
+task('reclaimFunds')
+    .addPositionalParam('address', 'contract address')
+    .addPositionalParam('payoutAddress', 'reclaim funds to this address')
+    .setAction(async (args, hre) => {
+        await hre.run('compile');
+
+        let quiz = await hre.ethers.getContractAt('Quiz', args.address);
+        const tx = await quiz.reclaimFunds(args.payoutAddress);
+        const receipt = await tx.wait();
+        console.log(`Successfully reclaimed funds to ${args.payoutAddress}. Transaction hash: ${receipt!.hash}`);
+    });
+
+// Set gasless key-pair.
+task('setGaslessKeyPair')
+    .addPositionalParam('address', 'contract address')
+    .addPositionalParam('payerAddress', 'payer address')
+    .addPositionalParam('payerSecret', 'payer secret key')
+    .setAction(async (args, hre) => {
+        const quiz = await hre.ethers.getContractAt('Quiz', args.address);
+        const nonce = await hre.ethers.provider.getTransactionCount(args.payerAddress);
+        await setGaslessKeyPair(quiz, args.payerAddress, args.payerSecret, nonce);
+    });
+
+// Fetch image from IPFS.
+task('fetchImageFromIpfs')
+    .addPositionalParam('ipfsHash', 'IPFS hash of the image')
+    .addPositionalParam('outputPath', 'Path to save the fetched image')
+    .setAction(async (args, hre) => {
+        try {
+            const response = await axios.get(`https://ipfs.io/ipfs/${args.ipfsHash}`, {
+                responseType: 'arraybuffer'
+            });
+
+            fs.writeFileSync(args.outputPath, Buffer.from(response.data));
+            console.log(`Image fetched from IPFS and saved to ${args.outputPath}`);
+        } catch (error) {
+            console.error('Error fetching image from IPFS:', error);
+        }
+    });
+
+task('deployAndSetupAll')
+    .addOptionalParam('questionsFile', 'File containing questions in JSON format')
+    .addOptionalParam('couponsFile', 'File containing coupons, one per line')
+    .addOptionalParam('metadataFile', 'NFT metadata file path')
+    .addOptionalParam('reward', 'Reward in ROSE')
+    .addOptionalParam('gaslessAddress', 'Sender address for gasless transactions')
+    .addOptionalParam('gaslessSecret', 'Sender secret key for gasless transactions')
+    .addOptionalParam('fundGaslessAmount', 'Amount in ROSE to fund the gasless account')
+    .addOptionalParam('fundAmount', 'Amount in ROSE to fund the contract')
```

Comment:
Let's ditch those and use config only.

```suggestion
    .addPositionalParam('config', 'Quiz configuration filename')
```

Reply:
Removed optional parameters.